llvm/include/llvm/Transforms/Utils/SimplifyIndVar.h

```diff
@@ -52,12 +52,11 @@ class IVVisitor {
 /// where the first entry indicates that the function makes changes and the
 /// second entry indicates that it introduced new opportunities for loop
 /// unswitching.
-std::pair<bool, bool> simplifyUsersOfIV(PHINode *CurrIV, ScalarEvolution *SE,
-                                        DominatorTree *DT, LoopInfo *LI,
-                                        const TargetTransformInfo *TTI,
-                                        SmallVectorImpl<WeakTrackingVH> &Dead,
-                                        SCEVExpander &Rewriter,
-                                        IVVisitor *V = nullptr);
+std::pair<bool, bool>
+simplifyUsersOfIV(PHINode *CurrIV, ScalarEvolution *SE, DominatorTree *DT,
+                  LoopInfo *LI, const TargetTransformInfo *TTI,
+                  SmallVectorImpl<WeakTrackingVH> &Dead, SCEVExpander &Rewriter,
+                  unsigned MaxDepthOutOfLoop = 1, IVVisitor *V = nullptr);
```

Comment:
Do we need the MaxDepthOutOfLoop parameter? Can you move the cl::opt into SimplifyIndVar and directly use it instead?

Reply:
`Transforms/Utils/LoopUnroll.cpp` uses it via `simplifyLoopIVs`. I didn't want to add it here as it could create regressions in passes I'm not interested in.

Reply:
Okay, that makes sense. But as implemented, aren't you still calling this with the default of 1 from LoopUnroll? Shouldn't the default be 0 instead?

Could you please rebase over https://github.com/llvm/llvm-project/commit/6e3725d7f290d0180e8b5cfe073d0cc4c9bbafd0 so we at least don't have to pass through this parameter everywhere?

Reply:
The question you raise is actually quite important as actually this code is disabled with this current default config. I disabled it as I wanted to make it optimisation level dependent but I needed some input to calibrate it according to the level.

It is set to `1` which means the set of BBs that will be affected are `dist(Loop, BB) < 1` (the strict bound is on purpose). The `0` value is a special value for an `+infinity` bound. 

+1 for making `pushIVUsers` a method instead of functions.

I think I'll put it to `0` now as it will show performance regressions and allow discussing about that. I'll add some tests before though.